beacon_node/eth2_libp2p/src/discovery/mod.rs

```diff
@@ -809,11 +770,19 @@ impl<TSpec: EthSpec> Discovery<TSpec> {
         let predicate: Box<dyn Fn(&Enr) -> bool + Send> =
             Box::new(move |enr: &Enr| eth2_fork_predicate(enr) && additional_predicate(enr));
 
+        // Indicate that we are starting a `FindPeers` query
+        if let QueryType::FindPeers = &query {
+            self.find_peer_active = true;
```

Comment:
I think it might be clearer and easy to track if we set this in the `discover_peers()` function just before we call start_query. I can't see a reason where we would ever not run a FindNode query after calling `start_query`. This way, this specific variable is localised to the `discover_peers()` function and we dont have to do this explicit match.